docs/spec/src/protocol/contracts.md

```diff
@@ -59,25 +126,184 @@ versionedBlobParams[0] = {
+**Description**
+
+Payment contract used to escrow on-demand funds, hold user reservations, and define global payment parameters used by the network (i.e, `globalSymbolsPerPeriod`, `reservationPeriodInterval`, `globalRatePeriodInterval`).
+
+**Access Mgmt**
+
+- `Ownable` role that can set payment reservations
+
+**Offchain Usage**
+
+TODO
+
+### [`PauserRegistry`](https://github.com/Layr-Labs/eigenlayer-contracts/blob/ac57bc1b28c83d9d7143c0da19167c148c3596a3/src/contracts/permissions/PauserRegistry.sol)
+
+**Description**
+Manages a stateful mapping of pausers that can be arbitrarily added or revoked. This contract is assumed to be deployed immutably. The pauser mapping is checked by caller:
+- Mapping checked as prerequisite for pausing batch confirmation logic in [`EigenDAServiceManager`](#eigendaservicemanager)
+- Mapping checked as prerequisite for pausing operator state update logic in [`RegistryCoordinator`](#eigendaregistrycoordinator)
+
+**Access Mgmt**
+- `Unpauser` (or admin) role that can set / remove existing pausers
+
+**Offchain Usage**
+
+TODO
+
+### [`BLSApkRegistry`](https://github.com/Layr-Labs/eigenlayer-middleware/blob/2f7c93e38f56f292f247981a52bd3619a16b9918/src/BLSApkRegistry.sol)
 
-<!-- TODO: we have this old doc https://www.notion.so/eigen-labs/EigenDA-V2-Governance-17513c11c3e0806999cfe5e8b9bf7e6a -->
-<!-- Not sure if it's still relevant or all outdated... will need to be written by @pakim249CAL once roles are stabilized. -->
+**Description**
+This contract stores each operator's BLS public key as well as per quorum aggregate public keys which are only updatable by the `RegistryCoordinator`.
+
+**Access Mgmt**
+- `RegistryCoordinator` role that can invoke aggregate key updates via the registration/de-registration of operators
+
+
+**Offchain Usage**
+
+TODO
+
+### [`IndexRegistry`](https://github.com/Layr-Labs/eigenlayer-middleware/blob/2f7c93e38f56f292f247981a52bd3619a16b9918/src/IndexRegistry.sol)
+
+**Description**
+Maintains an ordered, historically versioned list of operators for each quorum, allowing the RegistryCoordinator to register or deregister operators while preserving full block-by-block history of operator counts and index assignments. It provides efficient read functions to reconstruct the operator set at any block
+
+**Access Mgmt**
+- `RegistryCoordinator` role that makes stateful updates when registering / deregistering quorum operators
+
+**Offchain Usage**
+
+TODO
+
+### [`StakeRegistry`](https://github.com/Layr-Labs/eigenlayer-middleware/blob/2f7c93e38f56f292f247981a52bd3619a16b9918/src/StakeRegistry.sol)
+
+**Description**
+Stores stake updates bounded by block number and quorum strategy:
+```solidity
+    struct StakeUpdate {
+        // the block number at which the stake amounts were updated and stored
+        uint32 updateBlockNumber;
+        // the block number at which the *next update* occurred.
+        /// @notice This entry has the value **0** until another update takes place.
```

Comment:
do we need to add documentation for these middleware contracts? given that middlware repo already contains full docs that we point to already